### ISSUE TITLE
optional CFF / CFF2 parsing?

### ISSUE THREAD

**Issue by Pomax** [Pomax/Font.js#48]
The lib is not intended to be a shaper, but it might be useful to offer a separate CFF/CFF2 parser that can be used to dig into the CFF/CFF2 blocks without doing charstring interpretation, but simply exposing their data in the same "raw data" way the rest of the font is already exposed as.

**Comment by Pomax** [comment]
(note: font.js should still offer built in high-level deconstruction, but not operator/operand unpacking)

**Comment by hyvyys** [comment]
+1. Right now I could use being able to tell whether a glyph contains any outlines at all.

**Comment by Pomax** [comment]
Can you sketch a use-case? Are you trying to determine if a glyph is printable from font, rather than from charset spec?

**Comment by hyvyys** [comment]
Oh, it’s nothing really. I’m making a string generator to use with pattern fonts. Some of them will use a subset of the basic Latin alphabet for their pattern modules and have the remaining glyph slots empty instead of just leaving them out of the font. So being able to tell a glyph is empty would help automate the glyph selection process.

**Comment by Pomax** [comment]
That seems like solving the symptom rather than the problem: don't include empty glyphs in your subset fonts?

**Comment by hyvyys** [comment]
Obviously. But sometimes curing the symptoms is all the doctor can do. I’m not working with my own fonts, and as the goal is to use a still smaller selection of the existing functional glyphs at any given time, manual glyph selection is still involved. I just thought I would detect and pre-select the filled glyphs after loading the font so that the users that come with their own custom-made, ad-hoc, imperfect fonts have a few clicks less.

**Comment by Pomax** [comment]
I see. In that case, you're probably better off loading a full-fat font library like OpenType.js, and then checking each glyph's metrics. I've implemented several TTF and CFF parsers in the past, and it's enough of an absolute nonsense job that unless someone's willing to fund the effort, I'm unlikely to get to it until I personally need it again.
